Source/WebKit/WebProcess/WebStorage/StorageNamespaceImpl.cpp

```diff
@@ -81,7 +81,7 @@ void StorageNamespaceImpl::destroyStorageAreaMap(StorageAreaMap& map)
 Ref<StorageArea> StorageNamespaceImpl::storageArea(const SecurityOrigin& securityOrigin)
 {
     auto& map = m_storageAreaMaps.ensure(securityOrigin.data(), [&] {
-        return makeUnique<StorageAreaMap>(*this, securityOrigin);
+        return makeUniqueWithoutRefCountedCheck<StorageAreaMap>(*this, securityOrigin);
```

Comment:
Forwarding ref-counting here doesn't seem safe. There is code removing the StorageAreaMap from m_storageAreaMaps, which would destroy the StorageAreaMap even if someone was holding a RefPtr to it.

Reply:
Good point--should be fixed now!